File: src/foragainstoronthefence.lua
xp, yp = io.read("*n", "*n")
xv, yv = io.read("*n", "*n")
u = io.read("*n")
d = (xp - xv) * (xp - xv) + (yp - yv) * (yp - yv)
if d == u then
    print("on the fence")
elseif d < u then
    print("for")
else
    print("against")
end

File: src/foragainstoronthefence.py
xp, yp = [int(d) for d in input().split()]
xv, yv = [int(d) for d in input().split()]
u = int(input())
d = (xp - xv) ** 2 + (yp - yv) ** 2
if d == u:
    print("on the fence")
elif d < u:
    print("for")
else:
    print("against")
